larray/core/array.py

```diff
@@ -5435,15 +5360,105 @@ def equals(self, other, rtol=0, atol=0, nan_equals=False):
         >>> # The reason is that a nan value is different from *anything*, including itself.
         >>> arr1.equals(arr2)
         False
-        >>> # set flag nan_equals to True to overwrite this behavior
-        >>> arr1.equals(arr2, nan_equals=True)
+        >>> # set flag nans_equal to True to overwrite this behavior
+        >>> arr1.equals(arr2, nans_equal=True)
         True
         """
         try:
             other = aslarray(other)
         except Exception:
             return False
-        return self.axes == other.axes and all(element_equal(self, other, rtol=rtol, atol=atol, nan_equals=nan_equals))
+        return self.axes == other.axes and all(self.eq(other, rtol=rtol, atol=atol, nans_equal=nans_equal))
+
+    @deprecate_kwarg('nan_equals', 'nans_equal')
+    def eq(self, other, rtol=0, atol=0, nans_equal=False):
+        """
+        Compares self with another array element-wise and returns an array of booleans.
+
+        Parameters
+        ----------
+        other: LArray-like
+            Input array. aslarray() is used on a non-LArray input.
+        rtol : float or int, optional
+            The relative tolerance parameter (see Notes). Defaults to 0.
+        atol : float or int, optional
+            The absolute tolerance parameter (see Notes). Defaults to 0.
+        nans_equal: boolean, optional
+            Whether or not to consider nan values at the same positions in the two arrays as equal.
+            By default, an array containing nan values is never equal to another array, even if that other array
+            also contains nan values at the same positions. The reason is that a nan value is different from
+            *anything*, including itself. Defaults to False.
+
+        Returns
+        -------
+        LArray
+            Boolean array of where corresponding elements of self and other are equal
```

Comment:
"of where" is not correct English. "whether" would already be better than "where" but I am unsure "of whether" is correct either. I can't come up with any phrasing I am very enthusiastic about but below are a few I think are correct even if not elegant:

* Boolean array telling whether corresponding elements of self and other are equal...
* Boolean array where each cell tells whether corresponding elements of self and other are equal...
* Boolean array resulting from evaluating whether corresponding elements of self and other are equal...